### PR TITLE
[kubernetes] Kubernetes C Client

### PR DESCRIPTION
**Pull request by ahmedyarub**
This adds Kubernetes C Client library

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by cenit** [comment]
two question:
1) usage should not be required. Is vcpkg unable to write automatically the same thing as you wrote, inspecting the target the lib creates?
2) why vcpkg_from_git and not vcpkg_from_github? the latter should be preferred for github repo

**Comment by ahmedyarub** [comment]
This PR fixes the MacOS errors: https://github.com/kubernetes-client/c/pull/84
I'll update this PR once that is merged.

**Comment by ahmedyarub** [comment]
> two question:
> 
> 1. usage should not be required. Is vcpkg unable to write automatically the same thing as you wrote, inspecting the target the lib creates?
> 2. why vcpkg_from_git and not vcpkg_from_github? the latter should be preferred for github repo

Done and done. Thank you.

**Comment by ahmedyarub** [comment]
MacOS version fixed.

File: ports/kubernetes/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "kubernetes",
+  "version-string": "0.0.1",
```
Comment (PhoebeHui) [inline ports/kubernetes/vcpkg.json:3]:
```suggestion
  "version": "0.0.1",
```

File: ports/kubernetes/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    REF ecc9efc23656e747e97f86fbe2810f7e8471990f
+    SHA512 e9bb1b6f68aacd82df2b33fcb432fdfee5fb56e0426835f8480afff1c2545f483eb0f4edeafb2e866ea74918d748aa48655b5819909c13a88f693e371047e8b7
+    HEAD_REF master
+    PATCHES
+        001-fix-destination.patch
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}/kubernetes
+    PREFER_NINJA)
```
Comment (PhoebeHui) [inline ports/kubernetes/portfile.cmake:15]:
```suggestion
)
```

File: ports/kubernetes/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  "description": "Kubernetes C client",
+  "homepage": "https://github.com/kubernetes-client/c/",
```
Comment (PhoebeHui) [inline ports/kubernetes/vcpkg.json:5]:
```suggestion
  "homepage": "https://github.com/kubernetes-client/c/",
  "supports": "!uwp",
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@ahmedyarub, thanks for the PR!

**Review by PhoebeHui — APPROVED** [review]
Thanks for your updates!

**Comment by strega-nil-ms** [comment]
Cool, Thanks @ahmedyarub !

**Pull request merged**